### PR TITLE
Risk Manager contract

### PR DESCRIPTION
**Pull request by grujawork**

### CODE REVIEW

File: contracts/SportMarkets/SportAMMRiskManager.sol [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    /// @notice Setting the Cap per Sport ID
+    /// @param _sportID The tagID used for sport (9004)
+    /// @param _childID The tagID used for childid (10002)
+    /// @param _capPerChild The cap amount used for the sportID
+    function setCapPerSportAndChild(
+        uint _sportID,
+        uint _childID,
+        uint _capPerChild
+    ) external onlyOwner {
+        capPerSportAndChild[_sportID][_childID] = _capPerChild;
```
Comment (danijelthales) [inline contracts/SportMarkets/SportAMMRiskManager.sol:141]: Check value of all params for validity.
Ensure capPerChild is not higher than capPerParent.

File: contracts/SportMarkets/SportAMMRiskManager.sol [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+        emit SetCapPerSportAndChild(_sportID, _childID, _capPerChild);
+    }
+
+    /// @notice Setting the Cap per Sport ID
+    /// @param _sportID The tagID used for each market
+    /// @param _capPerSport The cap amount used for the sportID
+    function setCapPerSport(uint _sportID, uint _capPerSport) external onlyOwner {
+        capPerSport[_sportID] = _capPerSport;
```
Comment (danijelthales) [inline contracts/SportMarkets/SportAMMRiskManager.sol:149]: Check value of all params for validity.

File: contracts/SportMarkets/SportAMMRiskManager.sol [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        emit SetCapPerSport(_sportID, _capPerSport);
+    }
+
+    /// @notice Setting the Cap per game default value
+    /// @param _capPerGame default cap
+    function setDefaultCapPerGame(uint _capPerGame) external onlyOwner {
+        defaultCapPerGame = _capPerGame;
```
Comment (danijelthales) [inline contracts/SportMarkets/SportAMMRiskManager.sol:156]: I generally think we need to change this value from 5000 to 1000.
Also maybe introduce a MaxCap variable e.g. 20k and ensure no value is greater than it.

File: contracts/SportMarkets/SportAMMRiskManager.sol [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+        emit SetDefaultCapPerGame(_capPerGame);
+    }
+
+    /// @notice default risk multiplier
+    /// @param _riskMultiplier risk multiplier
+    function setDefaultRiskMultiplier(uint _riskMultiplier) external onlyOwner {
+        defaultRiskMultiplier = _riskMultiplier;
```
Comment (danijelthales) [inline contracts/SportMarkets/SportAMMRiskManager.sol:163]: I would like to introduce maxRisk multiplier variable, e.g. 5 and require that the multiplier here is not higher than that.

File: contracts/SportMarkets/SportAMMRiskManager.sol [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+        );
+        for (uint i; i < _markets.length; i++) {
+            riskMultiplierPerMarket[_markets[i]] = _riskMultiplier;
+            emit SetRiskMultiplierPerMarket(_markets[i], _riskMultiplier);
+        }
+    }
+
+    /// @notice Setting the Sport Positional Manager contract address
+    /// @param _manager Address of Staking contract
+    function setSportsPositionalMarketManager(address _manager) external onlyOwner {
```
Comment (danijelthales) [inline contracts/SportMarkets/SportAMMRiskManager.sol:191]: check for 0 address?

File: contracts/SportMarkets/SportAMMRiskManager.sol [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+        tag1 = sportMarket.tags(0);
+        tag2 = sportMarket.isChild() ? sportMarket.tags(1) : 0;
+    }
+
+    /* ========== CONTRACT MANAGEMENT ========== */
+
+    /// @notice Setting the Cap per spec. market
+    /// @param _markets market addresses
+    /// @param _capPerMarket The cap amount used for the specific markets
+    function setCapPerMarket(address[] memory _markets, uint _capPerMarket) external {
```
Comment (danijelthales) [inline contracts/SportMarkets/SportAMMRiskManager.sol:121]: I would like to have a max cap variable of e.g. 20k.

File: contracts/SportMarkets/SportAMMRiskManager.sol (hunk 1 above)
Comment (grujawork) [reply contracts/SportMarkets/SportAMMRiskManager.sol:154]: done.

File: contracts/SportMarkets/SportAMMRiskManager.sol (hunk 2 above)
Comment (grujawork) [reply contracts/SportMarkets/SportAMMRiskManager.sol:164]: done.

File: contracts/SportMarkets/SportAMMRiskManager.sol (hunk 3 above)
Comment (grujawork) [reply contracts/SportMarkets/SportAMMRiskManager.sol:172]: done. put it in script 1K

File: contracts/SportMarkets/SportAMMRiskManager.sol (hunk 4 above)
Comment (grujawork) [reply contracts/SportMarkets/SportAMMRiskManager.sol:180]: done.

File: contracts/SportMarkets/SportAMMRiskManager.sol (hunk 5 above)
Comment (grujawork) [reply contracts/SportMarkets/SportAMMRiskManager.sol:212]: done.

File: contracts/SportMarkets/SportAMMRiskManager.sol (hunk 6 above)
Comment (grujawork) [reply contracts/SportMarkets/SportAMMRiskManager.sol:128]: done.

File: contracts/SportMarkets/SportAMMRiskManager.sol [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+    }
+
+    /// @notice returns if market is in to much of a risk
+    /// @param _totalSpent total spent on market
+    /// @param _market for which is calculation done
+    /// @return _isNotRisky true/false
+    function isTotalSpendingLessThanTotalRisk(uint _totalSpent, address _market) external view returns (bool _isNotRisky) {
+        uint capPerMarket = _calculateCapToBeUsed(_market);
+        uint riskMultiplier = _calculateRiskMultiplier(_market);
+        return _totalSpent < capPerMarket * riskMultiplier;
```
Comment (danijelthales) [inline contracts/SportMarkets/SportAMMRiskManager.sol:85]: less or equal might be better

**Review by danijelthales — APPROVED** [review]

File: contracts/SportMarkets/SportAMMRiskManager.sol (hunk 7 above)
Comment (grujawork) [reply contracts/SportMarkets/SportAMMRiskManager.sol]: done.

**Review by vpoklopic — APPROVED** [review]

File: contracts/SportMarkets/SportsAMM.sol [hunk 8]
```diff
@@ -683,7 +683,8 @@ contract SportsAMM is Initializable, ProxyOwned, PausableUpgradeable, ProxyReent
         address _referrals,
         address _parlayAMM,
         address _wrapper,
-        address _lp
+        address _lp,
+        address _riskManager
```
Comment (kirilaa) [inline contracts/SportMarkets/SportsAMM.sol:687]: Not a fan of extending this function due to pDAO upgrades, but let's stop with the risk manager, as a last one. 
It's 9 attempts to make a copy-paste mistake

Comment (grujawork) [reply contracts/SportMarkets/SportsAMM.sol:687]: true

**Review by kirilaa — APPROVED** [review]

**Pull request merged**